lib/evmone/instructions.hpp

```diff
@@ -1052,22 +1265,26 @@ inline Result log(StackTop stack, int64_t gas_left, ExecutionState& state) noexc
     if (state.in_static_mode())
         return {EVMC_STATIC_MODE_VIOLATION, 0};
 
-    const auto& offset = stack.pop();
-    const auto& size = stack.pop();
+    const auto& offset = stack[0];
+    const auto& size = stack[1];
 
-    if (!check_memory(gas_left, state.memory, offset, size))
+    if (!check_memory(gas_left, state.memory, offset.val, size.val))
         return {EVMC_OUT_OF_GAS, gas_left};
 
-    const auto o = static_cast<size_t>(offset);
-    const auto s = static_cast<size_t>(size);
+    const auto o = static_cast<size_t>(offset.val);
+    const auto s = static_cast<size_t>(size.val);
 
     const auto cost = int64_t(s) * 8;
     if ((gas_left -= cost) < 0)
         return {EVMC_OUT_OF_GAS, gas_left};
 
     std::array<evmc::bytes32, NumTopics> topics;  // NOLINT(cppcoreguidelines-pro-type-member-init)
-    for (auto& topic : topics)
-        topic = intx::be::store<evmc::bytes32>(stack.pop());
+    int topic_counter = 2;
+    for (auto& topic : topics){
+        topic = intx::be::store<evmc::bytes32>(stack[topic_counter].val);
```

Comment:
With the current implementation, I think we need exact match requirement on all the topics here. I think it will be possible to relax the exact match, but probably required changes to the host interface.